src/components/other/ProjectAddForm/index.js

```diff
@@ -58,142 +60,131 @@ const mapDispatchToProps = dispatch => ({
-                error={faramErrors}
-                disabled={pending}
-            >
-                { pending && <LoadingAnimation /> }
-                <NonFieldErrors faramElement />
-                <TextInput
-                    faramElementName="title"
-                    label={_ts('components.addProject', 'addProjectModalLabel')}
-                    placeholder={_ts('components.addProject', 'addProjectModalPlaceholder')}
-                    autoFocus
-                />
-                <Cloak
-                    hide={ProjectAddForm.shouldHidePrivate}
-                    render={
-                        <SegmentInput
-                            options={projectVisibilityOptions}
-                            className={styles.isPrivateCheckbox}
-                            faramElementName="isPrivate"
-                            label={_ts('components.addProject', 'projectVisibilityInputLabel')}
-                            hint={_ts('components.addProject', 'projectVisibilityInputHint')}
-                        />
-                    }
-                />
-                <div className={styles.actionButtons}>
-                    <DangerButton onClick={this.onModalClose}>
-                        {_ts('components.addProject', 'modalCancel')}
-                    </DangerButton>
-                    <PrimaryButton
-                        type="submit"
-                        disabled={pending || !pristine}
-                    >
-                        {_ts('components.addProject', 'modalCreate')}
-                    </PrimaryButton>
-                </div>
-            </Faram>
-        );
-    }
+const projectAddSchema = {
+    fields: {
+        title: [requiredCondition],
+        isPrivate: [],
+        organizations: [],
+    },
+};
+
+function ProjectAddForm(props) {
+    const {
+        className,
+        onModalClose,
+        userGroups,
+        userId,
+        onProjectAdd,
+        setUserProject,
+        setUsergroupProject,
+        setUserProfileProject,
+    } = props;
+
+    const [faramValues, setFaramValues] = useState({ isPrivate: false, organizations: [] });
+    const [faramErrors, setFaramErrors] = useState({});
+    const [pristine, setPristine] = useState(true);
+
+    const [
+        pending,
+        ,
+        ,
+        createNewProject,
+    ] = useRequest({
+        url: 'server://projects/',
+        body: {
+            title: faramValues.title,
+            userGroups,
+            isPrivate: faramValues.isPrivate,
+            organizations: faramValues.organizations,
+        },
+        method: 'POST',
+        onSuccess: (response) => {
+            setUserProject({ project: response });
+            setUsergroupProject({ project: response });
+            setUserProfileProject({
+                userId,
+                project: response,
+            });
+            if (onProjectAdd) {
+                onProjectAdd(response.id);
+            }
```

Comment:
Is this correct? Shouldn't onProjectAdd accept `response` then the parent handle setUserProject, setUsergroupProject and setUserProfileProject on the parent itself?

Reply:
This was done previously and I replicated it